classes/class-wc-connect-label-reports.php

```diff
@@ -0,0 +1,163 @@
+			$db_results = $wpdb->get_results( $query );
+			$results = array();
+
+			foreach( $db_results as $meta ) {
+				$labels = maybe_unserialize( $meta->meta_value );
+
+				if ( ! is_array( $labels ) ) {
+					$labels = $this->settings_store->try_deserialize_labels_json( $meta->meta_value );
+				}
+
+				if ( empty( $labels ) ) {
+					continue;
+				}
+
+				foreach ( $labels as $label ) {
+					$results[] = array_merge( $label, array( 'order_id' => $meta->post_id ) );
+				}
+			}
+
+			usort( $results, array( $this, 'compare_label_dates_desc' ) );
+			return $results;
+		}
+
+		private function query_labels() {
+			$all_labels = $this->get_all_labels();
+
+			// translate timestamps to JS timestapms
+			$start_date = $this->start_date * 1000;
+			$end_date = $this->end_date * 1000;
+
+			$results = array();
+			foreach ( $all_labels as $label ) {
+				$created = $label['created'];
+				if ( $created > $end_date ) {
+					continue;
+				}
+
+				//labels are sorted in descending order, so if we reached the end, break the loop
+				if ( $created < $start_date ) {
+					break;
+				}
+
+				if ( isset( $label['error'] ) || ! isset( $label['rate'] ) ) {
+					continue;
+				}
+
+				$results[] = $label;
+			}
+
+			return $results;
+		}
+
+		public function output_report() {
+			$ranges = array(
+				'year'       => __( 'Year', 'woocommerce-services' ),
+				'last_month' => __( 'Last month', 'woocommerce-services' ),
+				'month'      => __( 'This month', 'woocommerce-services' ),
+				'7day'       => __( 'Last 7 days', 'woocommerce-services' ),
+			);
+
+			$current_range = ! empty( $_GET['range'] ) ? sanitize_text_field( $_GET['range'] ) : '7day';
+
+			if ( ! in_array( $current_range, array( 'custom', 'year', 'last_month', 'month', '7day' ) ) ) {
+				$current_range = '7day';
+			}
+
+			$this->check_current_range_nonce( $current_range );
+			$this->calculate_current_range( $current_range );
+
+			$hide_sidebar = true;
+
+			include( WC()->plugin_path() . '/includes/admin/views/html-report-by-date.php' );
+		}
+
+		/**
+		 * Get the main chart.
+		 */
+		public function get_main_chart() {
+			$labels = $this->query_labels();
+
+			?>
+			<table class="widefat">
+				<thead>
+					<tr>
+						<th><?php esc_html_e( 'Time', 'woocommerce-services' ); ?></th>
+						<th><?php esc_html_e( 'Order', 'woocommerce-services' ); ?></th>
+						<th><?php esc_html_e( 'Price', 'woocommerce-services' ); ?></th>
+						<th><?php esc_html_e( 'Service', 'woocommerce-services' ); ?></th>
+					</tr>
+				</thead>
+				<?php if ( ! empty( $labels ) ) : ?>
+					<tbody>
+						<?php
+						foreach ( $labels as $label ) {
+							?>
+							<tr>
+								<th scope="row"><?php echo date_i18n( get_option( 'date_format' ) . ' g:i a', $label['created'] / 1000 ); ?></th>
+								<td><a href="<?php echo admin_url( 'post.php?post=' . $label['order_id'] . '&action=edit' ); ?>"><?php echo $label['order_id']; ?></a></td>
```

Comment:
There are extensions that allow merchants to change their order numbers ([#](https://woocommerce.com/products/sequential-order-numbers-pro/), [#](https://booster.io/features/woocommerce-custom-order-numbers/)) - I think we'll need to retrieve the `WC_Order` so that we're displaying the correct order number here (it doesn't always match the post ID).